exchanges/cryptodotcom/cryptodotcom_types.go

```diff
@@ -0,0 +1,703 @@
+		BaseCurrencySize  float64              `json:"base_currency_size"`
+		QuoteCurrencySize float64              `json:"quote_currency_size,string"`
+		QuoteBuy          float64              `json:"quote_buy,string"`
+		QuoteSell         float64              `json:"quote_sell,string"`
+		QuoteDuration     int64                `json:"quote_duration"`
+		QuoteTime         cryptoDotComMilliSec `json:"quote_time"`
+		QuoteExpiryTime   cryptoDotComMilliSec `json:"quote_expiry_time"`
+		TradeDirection    string               `json:"trade_direction"`
+		TradePrice        float64              `json:"trade_price"`
+		TradeQuantity     float64              `json:"trade_quantity"`
+		TradeValue        float64              `json:"trade_value"`
+		TradeTime         cryptoDotComMilliSec `json:"trade_time"`
+	} `json:"quote_list"`
+}
+
+// OTCTradeHistoryResponse represents an OTC trade history response.
+type OTCTradeHistoryResponse struct {
+	Count     int64          `json:"count"`
+	TradeList []OTCTradeItem `json:"trade_list"`
+}
+
+// OTCTradeItem represents an OTC trade item detail.
+type OTCTradeItem struct {
+	QuoteID           string               `json:"quote_id"`
+	QuoteStatus       string               `json:"quote_status"`
+	QuoteDirection    string               `json:"quote_direction"`
+	BaseCurrency      string               `json:"base_currency"`
+	QuoteCurrency     string               `json:"quote_currency"`
+	BaseCurrencySize  float64              `json:"base_currency_size,string"`
+	QuoteCurrencySize float64              `json:"quote_currency_size,string"`
+	QuoteBuy          string               `json:"quote_buy"`
+	QuoteSell         string               `json:"quote_sell"`
+	QuoteDuration     int64                `json:"quote_duration"`
+	QuoteTime         cryptoDotComMilliSec `json:"quote_time"`
+	QuoteExpiryTime   cryptoDotComMilliSec `json:"quote_expiry_time"`
+	TradeDirection    string               `json:"trade_direction"`
+	TradePrice        float64              `json:"trade_price,string"`
+	TradeQuantity     float64              `json:"trade_quantity,string"`
+	TradeValue        float64              `json:"trade_value,string"`
+	TradeTime         cryptoDotComMilliSec `json:"trade_time"`
+}
+
+// SubscriptionPayload represents a subscription payload
+type SubscriptionPayload struct {
+	ID            int64               `json:"id"`
+	Method        string              `json:"method"`
+	Params        map[string][]string `json:"params"`
+	Nonce         int64               `json:"nonce"`
+	Authenticated bool                `json:"-"`
+}
+
+// SubscriptionResponse represents a websocket subscription response.
+type SubscriptionResponse struct {
+	ID     int64    `json:"id"`
+	Code   int64    `json:"code,omitempty"`
+	Method string   `json:"method"`
+	Result WsResult `json:"result"`
+}
+
+// SubscriptionRawData represents a subscription response raw data.
+type SubscriptionRawData struct {
+	Data          stream.Response
```

Comment:
Just use `[]byte` we don't need to keep allocating `stream.Response` every data event